scripts/check_urls.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/python3
+
+import os
+
+def scanDirectory(path):
+    entries = os.listdir(path)
+    for entry in entries:
+        if os.path.isfile(os.path.join(path, entry)) and ".md" in entry:
+            readFile(os.path.join(path, entry))
+        if (os.path.isdir(os.path.join(path, entry))):
+            scanDirectory(os.path.join(path, entry))
+
+import re
+import requests
+import multiprocessing as mp
+
+manager = mp.Manager()
+invalid_uris = manager.list()
+uris = []
+
+
+def readFile(path):
+    with open(path, 'r') as f:
+        for line in f.readlines():
+            result = re.search(r"\[(.*?)\]\((.*?)\)", line)
+            if(result and not "ioit://" in result.group(2)):
+                uris.append(result.group(2))
+
+def testhttpuri(uri):
+    try:
+        headers = {
+            'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36'
+        }
+        r = requests.get(uri, headers=headers, timeout = 5)
+        if r.status_code != requests.codes.ok:
+            print(uri,r.status_code)
+            invalid_uris.append(uri)
+    except:
+        print("failed to connect")
+        invalid_uris.append(uri)
+        
+scanDirectory("./locales")
```

Comment:
in this way, are we testing the same urls twice? a set for italian and the same set for english?
if it is, it could be increase if we'll support another language

Reply:
@Undermaken 
Yes, we check it in both italian and english sub-dir, i decided to use this approach cause it happened to have the links wrongly written in one language but not in the other one

Reply:
@CrisTofani 
Yes, good point!
What about making this change as improvement: at the beginning the script parse the folders (without forking any threads) and put all urls found in a [set](https://docs.python.org/3.8/library/stdtypes.html#set-types-set-frozenset) (to avoid duplicate)
Then all set items are splitted and each thread take its own part to parse.
What do you think?